lib/core/src/chain_swap.rs

```diff
@@ -1122,35 +1124,56 @@ impl ChainSwapHandler {
         swap_update_tx: &SwapUpdateTxDetails,
         verify_confirmation: bool,
     ) -> Result<()> {
-        match chain_swap.direction {
-            Direction::Incoming => {
-                self.verify_incoming_server_lockup_tx(
-                    chain_swap,
-                    swap_update_tx,
-                    verify_confirmation,
-                )
-                .await
-            }
-            Direction::Outgoing => {
-                self.verify_outgoing_server_lockup_tx(
-                    chain_swap,
-                    swap_update_tx,
-                    verify_confirmation,
-                )
-                .await
+        let swap_script = chain_swap.get_claim_swap_script()?;
+        let claim_details = chain_swap.get_boltz_create_response()?.claim_details;
+        let retries = if verify_confirmation { 30 } else { 0 };
+        let mut retry = 0;
+        loop {
```

Comment:
The concern here is that it will block the event status polling loop and may block other payment statuses.

Reply:
I think it would take a only a couple of attempts to get the confirmation, it testing it was confirmed in the first request. Maybe it's better to add it to the tracking thread to verify and claim if needed